File: datasets/lua-keep-incoder_6b_temp_0.2-3/HumanEval_104_unique_digits.lua
-- Given a list of positive integers x. return a sorted list of all 
-- elements that hasn't any even digit.
-- Note: Returned list should be sorted in increasing order.
-- For example:
-- >>> unique_digits([15, 33, 1422, 1])
-- [1, 15, 33]
-- >>> unique_digits([152, 323, 1422, 10])
-- []
local function unique_digits(x)
  local function helper(y)
    local i = 1
    while i <= y do
      if x[i] == 0 then
        table.insert(res, x[i])
      else
        helper(i * 10)
      end
      i = i + 1
    end
  end
  local res = {}
  helper(1)
  return res
end

-- Given a list of positive integers x. return a sorted list of all 
-- elements that hasn't any odd digit.
-- Note: Returned list should be sorted in increasing order.
-- For example:
-- >>> unique_digits([15, 33, 1422, 1])
-- [1, 33, 15]
-- >>> unique_digits([152, 323, 1422, 10])
-- []

lu = require('luaunit')

function test_humaneval()
local candidate = unique_digits
    lu.assertEquals(candidate({15, 33, 1422, 1}), {1, 15, 33})
    lu.assertEquals(candidate({152, 323, 1422, 10}), {})
    lu.assertEquals(candidate({12345, 2033, 111, 151}), {111, 151})
    lu.assertEquals(candidate({135, 103, 31}), {31, 135})
end

os.exit(lu.LuaUnit.run())

File: datasets/js-keep-code_davinci_001_temp_0.2/HumanEval_104_unique_digits.js
//Given a list of positive integers x. return a sorted list of all 
// elements that hasn't any even digit.
// Note: Returned list should be sorted in increasing order.
// For example:
// >>> unique_digits([15, 33, 1422, 1])
// [1, 15, 33]
// >>> unique_digits([152, 323, 1422, 10])
// []
function unique_digits(x){
    var result = [];
    for(var i = 0; i < x.length; i++){
        var num = x[i];
        var numStr = num.toString();
        var numArr = numStr.split('');
        var numArr2 = [];
        for(var j = 0; j < numArr.length; j++){
            var num2 = Number(numArr[j]);
            if(num2 % 2 != 0){
                numArr2.push(num2);
            }
        }
        if(numArr2.length == numArr.length){
            result.push(num);
        }
    }
    return result;
}


const assert = require('node:assert');


function test() {
  let candidate = unique_digits;
  assert.deepEqual(candidate([15, 33, 1422, 1]),[1, 15, 33]);
  assert.deepEqual(candidate([152, 323, 1422, 10]),[]);
  assert.deepEqual(candidate([12345, 2033, 111, 151]),[111, 151]);
  assert.deepEqual(candidate([135, 103, 31]),[31, 135]);
}

test();